flow/layers/physical_model_layer.cc

```diff
@@ -17,19 +22,32 @@ void PhysicalModelLayer::Preroll(PrerollContext* context, const SkMatrix& matrix
   PrerollChildren(context, matrix);
   if (!context->child_paint_bounds.intersect(rrect_.getBounds()))
     context->child_paint_bounds.setEmpty();
+  set_paint_bounds(rrect_bounds());
+}
 
-  // Add some margin to the paint bounds to leave space for the shadow.
-  // The margin is hardcoded to an arbitrary maximum for now because Skia
-  // doesn't provide a way to calculate it.
+// Add some margin to the paint bounds to leave space for the shadow.
+// The margin is hardcoded to an arbitrary maximum for now because Skia
+// doesn't provide a way to calculate it.
+SkRect PhysicalModelLayer::rrect_bounds() {
   SkRect bounds(rrect_.getBounds());
   bounds.outset(50.0, 50.0);
-  set_paint_bounds(bounds);
+  return bounds;
 }
 
+#if defined(OS_FUCHSIA)
+
+void PhysicalModelLayer::UpdateScene(SceneUpdateContext& context,
+                                     mozart::Node* container) {
+  context.AddLayerToCurrentPaintTask(this);
+  auto node = mozart::Node::New();
+  node->content_clip = mozart::RectF::From(rrect_bounds());
```

Comment:
We want to clip to `rrect_.getBounds()`, like we do in the non-mozart codepath.  The inflated bounds are to provide room for our shadow, but when clipping the children, we don't want to include the extra room.